lbrynet/file_manager/EncryptedFileManager.py

```diff
@@ -97,18 +97,21 @@ def _get_lbry_file(self, rowid, stream_hash, payment_rate_manager, sd_hash, key,
         )
 
     @defer.inlineCallbacks
-    def _start_lbry_file(self, file_info, payment_rate_manager):
+    def _start_lbry_file(self, file_info, payment_rate_manager, verify_stream, claim_info):
         lbry_file = self._get_lbry_file(
             file_info['row_id'], file_info['stream_hash'], payment_rate_manager, file_info['sd_hash'],
             file_info['key'], file_info['stream_name'], file_info['file_name'], file_info['download_directory'],
             file_info['suggested_file_name']
         )
-        yield lbry_file.get_claim_info()
+        if claim_info:
+            lbry_file.set_claim_info(claim_info)
         try:
-            # verify the stream is valid (we might have downloaded an invalid stream
-            # in the past when the validation check didn't work)
-            stream_info = yield get_sd_info(self.storage, file_info['stream_hash'], include_blobs=True)
-            validate_descriptor(stream_info)
+            # verify if the stream is valid (we might have downloaded an invalid stream
+            # in the past when the validation check didn't work. This runs after every
+            # migration to ensure blobs migrated from that past version gets verified)
+            if verify_stream:
+                stream_info = yield get_sd_info(self.storage, file_info['stream_hash'], include_blobs=True)
+                validate_descriptor(stream_info)
```

Comment:
I think this whole 

```
try:
    validate
except:
    delete invalid stream
```

block should be moved to a migration function in `lbrynet.database.migrator`

Validation of existing streams should occur just once rather than after each migration